pkg/cloudprovider/vsphere/loadbalancer/interface.go

```diff
@@ -0,0 +1,110 @@
+/*
+ Copyright 2020 The Kubernetes Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package loadbalancer
+
+import (
+	"github.com/vmware/go-vmware-nsxt/common"
+	"github.com/vmware/go-vmware-nsxt/loadbalancer"
+	clientset "k8s.io/client-go/kubernetes"
+	cloudprovider "k8s.io/cloud-provider"
+)
+
+// LoadBalancer is the interface used call the load balancer functionality
+// It extends the cloud controller manager LoadBalancer interface by an
+// initialization function
+type LoadBalancer interface {
+	cloudprovider.LoadBalancer
+	Initialize(client clientset.Interface, stop <-chan struct{})
+}
+
+// Access provides methods for dealing with NSX-T objects
+type Access interface {
```

Comment:
this is a lot like NsxtBroker interface. is there any significant benefit from this extra layer of abstraction?

Reply:
This was intended to have a hook for testing without real mocking. We had no time so far to continue with this.

Reply:
Can we give this a more specific name at least? `NSXTClient`?

Reply:
renamed